src/app/report/edit-display/edit-display.component.ts

```diff
@@ -85,22 +90,33 @@ export class EditDisplayComponent {
     private errorHandler: ErrorHandling,
   ) {}
 
-  showReport(report: Report): void {
+  showReport(node: ReportOrCheckpoint): void {
     this.disableEditing();
-    this.report = report;
-    report.xml
-      ? this.editor.setNewReport(report.xml)
-      : this.editor.setNewReport(this.helperService.convertMessage(report));
+    this.selectedNode = node;
+    if (ReportUtil.isReport(this.selectedNode)) {
+      this.editor.setNewReport(this.selectedNode.xml);
+    } else if (ReportUtil.isCheckPoint(this.selectedNode)) {
+      this.editor.setNewReport(this.convertMessage(this.selectedNode));
+    }
     this.rerunResult = undefined;
     this.displayReport = true;
   }
 
-  changeEncoding(button: any): void {
-    this.editor.setNewReport(this.helperService.changeEncoding(this.report, button));
+  convertMessage(checkpoint: Checkpoint): string {
+    let message: string = checkpoint.message;
+    if (checkpoint.encoding == 'Base64') {
+      message = btoa(message);
+    }
+    return message;
   }
 
   rerunReport(): void {
-    const reportId: number = this.report.storageId;
+    const node: ReportOrCheckpoint = this.selectedNode;
```

Comment:
I added this after you asked me to avoid a race condition/shared mutable state for the code. So I added a local variable so it avoids a shared mutable state in case selectedNode gets changed while the method is running.

Reply:
You're right, my bad.